### PR TITLE
fix: move usage reporting to cron jobs

### LINKED ISSUE #3373: Improve usage report with batch_num and max_ttl
### Which OpenObserve functionalities are relevant/related to the feature request?

_No response_

### Description

we do batch commit for usage report. but if we can't reach the limit it never report data, we some also add a time limit, even it didn't reach the batch_num, it already past x seconds, it also should be commit.

### Proposed solution

-

### Alternatives considered

-

### PR DESCRIPTION
**Pull request by Subhra264**
Fixes #3373 

## Summary by CodeRabbit

- **New Features**
  - Improved the handling and publishing of usage data, ensuring more efficient and timely updates.
  
- **Refactor**
  - Streamlined the internal control flow for managing usage data, enhancing overall system performance.

### CODE REVIEW

File: src/job/mod.rs [hunk 1]
```diff
@@ -95,6 +95,8 @@ pub async fn init() -> Result<(), anyhow::Error> {
         tokio::task::spawn(async move { telemetry::run().await });
     }
 
+    tokio::task::spawn(async move { usage::run().await });
```
Comment (Subhra264) [inline src/job/mod.rs:98]: @hengfeiyang should it run on all nodes? or I should leave compactor roles?

Comment (hengfeiyang) [reply src/job/mod.rs:98]: What is this job? accept usage event and send to ingester? Which node has the usage event? all the nodes or only some nodes?

File: src/service/usage/mod.rs [hunk 2]
```diff
@@ -363,9 +341,45 @@
...
+async fn publish_existing_triggers_usage() {
+    let mut usages = TRIGGERS_USAGE_DATA.write().await;
+    log::info!("publishing triggers usage reports,len: {}", usages.len());
+
+    let curr_usages = std::mem::take(&mut *usages);
+    // release the write lock
+    drop(usages);
+
+    ingest_trigger_usages(curr_usages).await
+}
```
[the remark this answers is not in the document]
Comment (hengfeiyang) [reply src/service/usage/mod.rs:366]: here, if the usages is empty, we can simple return.

File: src/job/mod.rs (hunk 1 above)
Comment (hengfeiyang) [reply src/job/mod.rs:98]: i am thinking we should run it on every node.

**Review by hengfeiyang — APPROVED** [review]

**Pull request merged**